File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/job/meta/StopJobRequest.scala
package com.webank.eggroll.core.deepspeed.job.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.constant.StringConstants
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class StopJobRequest(sessionId: String = StringConstants.EMPTY) {
}

object StopJobRequest {
  implicit def serialize(src: KillJobRequest): Array[Byte] = {
    Deepspeed.StopJobRequest.newBuilder()
      .setSessionId(src.sessionId)
      .build()
      .toByteArray
  }

  implicit def deserialize(byteString: ByteString): StopJobRequest = {
    val proto = Deepspeed.StopJobRequest.parseFrom(byteString)
    StopJobRequest(sessionId = proto.getSessionId)
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/StopJobRequest.java
package com.eggroll.core.pojo;

import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Data
public class StopJobRequest implements RpcMessage {

    Logger log = LoggerFactory.getLogger(StopJobRequest.class);
    private String sessionId;

    @Override
    public byte[] serialize() {
        Deepspeed.StopJobRequest.Builder builder = Deepspeed.StopJobRequest.newBuilder();
        builder.setSessionId(this.sessionId);
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {
        try {
            Deepspeed.StopJobRequest protoEntity = Deepspeed.StopJobRequest.parseFrom(data);
            this.sessionId = protoEntity.getSessionId();
        } catch (Exception e) {
            log.error("deserialize error : ", e);
        }
    }
}
